server/utils/parseDateTime.ts

```diff
@@ -0,0 +1,35 @@
+// NB: this module is shared with frontend code!
```

Comment:
perhaps this module could be names `utils/dateUtils`?

Reply:
There is date formatting in the `format` module so maybe that'd be too generic

Reply:
Interesting the `format` module is pretty much all to do with formatting dates and times 😅 

...while this one is all about parsing dates and times from user input, so I see why you named it `parseDateTime` 👍 (to me `parseDateTime` sounds almost like an utility function to parse datetimes IYKWIM rather than a noun thing but it's fine, having this in its own module makes sense and we can always rename in the future if we really want to)